core/src/main/kotlin/materialui/styles/withStyles.kt

```diff
@@ -11,16 +11,24 @@ private external val withStylesModule: dynamic
 @Suppress("UnsafeCastFromDynamic")
 private val withStyles: (dynamic, dynamic) -> ((Any) -> JsClass<*>) = withStylesModule.default
 
+fun <C : Component<P, *>, P : RProps> withStyles(
+    klazz: KClass<C>,
+    styleSet: StylesSet.() -> Unit,
+    withTheme: Boolean = false
+): RClass<P> {
+    return withStyles(
+        { theme: MuiTheme -> StylesBuilder(theme).apply(styleSet).toDynamic() },
+        js { this["withTheme"] = withTheme }
+    )(klazz.js) as RClass<P>
```

Comment:
`unsafeCast()` is more appropriate

```
withStyles(...)(klazz.js).unsafeCast<RClass<P>>()
```

Reply:
I agree. This was already existing code though. I will try to change as you described.